drivers/comparator/comparator_nrf_comp.c

```diff
@@ -0,0 +1,771 @@
+		shim_nrf_comp_data0.event_mask = NRF_COMP_INT_UP_MASK;
+		break;
+
+	case COMPARATOR_TRIGGER_FALLING_EDGE:
+		shim_nrf_comp_data0.event_mask = NRF_COMP_INT_DOWN_MASK;
+		break;
+
+	case COMPARATOR_TRIGGER_BOTH_EDGES:
+		shim_nrf_comp_data0.event_mask = NRF_COMP_INT_CROSS_MASK;
+		break;
+	}
+
+	if (shim_nrf_comp_is_resumed()) {
+		shim_nrf_comp_start();
+	}
+
+	return 0;
+}
+
+static int shim_nrf_comp_set_trigger_callback(const struct device *dev,
+					      comparator_callback_t callback,
+					      void *user_data)
+{
+	shim_nrf_comp_stop();
+
+	shim_nrf_comp_data0.callback = callback;
+	shim_nrf_comp_data0.user_data = user_data;
+
+	if (callback != NULL && atomic_test_and_clear_bit(&shim_nrf_comp_data0.triggered, 0)) {
+		callback(dev, user_data);
+	}
+
+	if (shim_nrf_comp_is_resumed()) {
+		shim_nrf_comp_start();
+	}
+
+	return 0;
+}
+
+static int shim_nrf_comp_trigger_is_pending(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return atomic_test_and_clear_bit(&shim_nrf_comp_data0.triggered, 0);
+}
+
+static const struct comparator_driver_api shim_nrf_comp_api = {
+	.get_output = shim_nrf_comp_get_output,
+	.set_trigger = shim_nrf_comp_set_trigger,
+	.set_trigger_callback = shim_nrf_comp_set_trigger_callback,
+	.trigger_is_pending = shim_nrf_comp_trigger_is_pending,
+};
+
+static int shim_nrf_comp_reconfigure(const nrfx_comp_config_t *nrf)
+{
+	shim_nrf_comp_stop();
+
+	(void)nrfx_comp_reconfigure(nrf);
+
+	if (shim_nrf_comp_is_resumed()) {
+		shim_nrf_comp_start();
+	}
+
+	return 0;
+}
+
+int comp_nrf_comp_configure_se(const struct device *dev,
+			       const struct comp_nrf_comp_se_config *config)
+{
+	nrfx_comp_config_t nrf = {};
+
+	ARG_UNUSED(dev);
+
+	if (shim_nrf_comp_se_config_to_nrf(config, &nrf)) {
+		return -EINVAL;
+	}
+
+	return shim_nrf_comp_reconfigure(&nrf);
+}
+
+int comp_nrf_comp_configure_diff(const struct device *dev,
+				 const struct comp_nrf_comp_diff_config *config)
+{
+	nrfx_comp_config_t nrf = {};
+
+	ARG_UNUSED(dev);
+
+	if (shim_nrf_comp_diff_config_to_nrf(config, &nrf)) {
+		return -EINVAL;
+	}
+
+	return shim_nrf_comp_reconfigure(&nrf);
+}
+
+static void shim_nrf_comp_event_handler(nrf_comp_event_t event)
+{
+	ARG_UNUSED(event);
+
+	if (shim_nrf_comp_data0.callback == NULL) {
+		atomic_set_bit(&shim_nrf_comp_data0.triggered, 0);
+		return;
+	}
+
+	shim_nrf_comp_data0.callback(DEVICE_DT_INST_GET(0), shim_nrf_comp_data0.user_data);
+	atomic_clear_bit(&shim_nrf_comp_data0.triggered, 0);
+}
+
+static int shim_nrf_comp_init(const struct device *dev)
+{
+	nrfx_comp_config_t nrf = {};
+
+	IRQ_CONNECT(DT_INST_IRQN(0),
+		    DT_INST_IRQ(0, priority),
+		    nrfx_isr,
+		    nrfx_comp_irq_handler,
+		    0);
+
+	irq_enable(DT_INST_IRQN(0));
+
+#if SHIM_NRF_COMP_DT_INST_MAIN_MODE_IS_SE(0)
+	(void)shim_nrf_comp_se_config_to_nrf(&shim_nrf_comp_config0, &nrf);
+#else
+	(void)shim_nrf_comp_diff_config_to_nrf(&shim_nrf_comp_config0, &nrf);
+#endif
+	(void)nrfx_comp_init(&nrf, shim_nrf_comp_event_handler);
```

Comment:
Updated to check the return value, looking through the `nrfx_comp_init` function, it can only fail on invalid state or failing to acquire something called `prs` which to my knowledge is not enabled for zephyr. I can check the value just to be safe, but really it should always return SUCCESS

Reply:
updated to check the value